### PR TITLE
Feature/dsnd 2086

### PR DESCRIPTION
**Pull request by maddytjCH**
* Added missing properties to the test application.properties
* Added missing ENV vars to the `maven-surefire-plugin` in POM
* Fixed the name and structure of ResourceChangedApiServiceAspect
* Minor refactoring and removing of unused imports

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/pscdataapi/service/CompanyPscService.java [hunk 1]
```diff
@@ -34,7 +34,6 @@
 import uk.gov.companieshouse.pscdataapi.api.ChsKafkaApiService;
 import uk.gov.companieshouse.pscdataapi.exceptions.BadRequestException;
 import uk.gov.companieshouse.pscdataapi.exceptions.ServiceUnavailableException;
-import uk.gov.companieshouse.pscdataapi.logging.DataMapHolder;
```
Comment (lbedrosyanCH) [inline src/main/java/uk/gov/companieshouse/pscdataapi/service/CompanyPscService.java:37]: This import is required - the project can't build without it

**Review by lbedrosyanCH — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+
+@ExtendWith(MockitoExtension.class)
+class ResourceChangedApiServiceAspectTest {
```
Comment (lbedrosyanCH) [inline src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceTest.java:12]: The class name has "Aspect", while the filename does not. They should be the same

File: src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagEnabledITest.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private ChangedResource changedResource;
+
+    private TestHelper testHelper;
+
+    @BeforeEach
+    void setup() {
+        testHelper = new TestHelper();
+    }
+    @Test
+    void testThatAspectShouldNotProceedWhenFeatureFlagEnabled() throws ServiceUnavailableException {
```
Comment (lbedrosyanCH) [inline src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagEnabledITest.java:46]: This test passes when ran using "mvn test", but it fails when ran on its own. Ideally we should set it up to pick up the correct environment variables even when ran on its own

File: src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagDisabledITest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+
+    @BeforeEach
+    void setup() {
+        when(internalApiClient.getHttpClient()).thenReturn(httpClient);
+        testHelper = new TestHelper();
+    }
+
+    @Test
+    void testThatKafkaApiShouldBeCalledWhenFeatureFlagDisabled()
```
Comment (lbedrosyanCH) [inline src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagDisabledITest.java:62]: I tried making the test fail by setting the feature flag to true in **feature.seeding_collection_enabled** in **application.properties**. I also tried adding **@ActiveProfiles("feature_flag_enabled")** to the class, similar to the other test class. However, the test still passed, which means we are not properly testing the feature flag

Comment (JAndrewCH) [reply src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagDisabledITest.java:62]: I'm happy that the testThatAspectShouldNotProceedWhenFeatureFlagEnabled test cover this test case

**Review by JAndrewCH — APPROVED** [review]

File: src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagDisabledITest.java (hunk 4 above)
Comment (maddytjCH) [reply src/test/java/uk/gov/companieshouse/pscdataapi/api/ResourceChangedApiServiceAspectFeatureFlagDisabledITest.java:62]: The @InjectMocks annotation bypasses the Aspect which means the Disabled test cannot fail, but the InjectMocks is required due to the structure of the API. The Enabled test does pass/fail when we want it to. Confirmation from James Andrew that we're happy to proceed with this.

**Pull request merged**